src/jaybenne/population_control.cpp

```diff
@@ -85,6 +87,24 @@ TaskStatus ControlPopulation(MeshData<Real> *md) {
         }
       });
 
+  //--------------------------------------------------------------------------------------
+  // print total number of active particles at a certain diagnostic level
+  if (diagnostic_level >= 1 && ncycle % ncycle_out == 0) {
+    Real num_tot_active_old = 0.0;
+    global_sum_reduce(
+        nblocks, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+        "ControlPopulation::report-tot-active-1",
+        [&](const int &b, const int &k, const int &j, const int &i, Real &tota) {
+          tota += vmesh(b, fj::active_num_per_cell(), k, j, i);
+        },
```

Comment:
```suggestion
       KOKKOS_LAMBDA(const int &b, const int &k, const int &j, const int &i, Real &tota) {
          tota += vmesh(b, fj::active_num_per_cell(), k, j, i);
        },
```
as you said in your comment. You've basically made an abstraction of the parthenon abstraction, so you would follow the same logic `par_reduce` does with respect to the kokkos `parallel_reduce`. Because of that, I'd also suggest moving the label to the first argument and passing the execution space here as the second argument.

Reply:
Done

Reply:
Oh wait - forgot to sort the arguments and add the execution space.

Reply:
Now the label is reordered (and actually used), and the `ExecSpace` is now a templated (2nd) argument to the function.